docs/integration_testing.rst

```diff
@@ -0,0 +1,157 @@
+fixture that represents the desired test run session results you're trying to validate. The ``Public Fixtures`` section
+of `conftest.py`_ contains all the shared fixtures meant to be used by test cases. Developers are encouraged to first
+scour the available fixtures to discover if one exists that covers the scenario that is under development for a given
+test case.
+
+Creating a Starting Fixture
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If an appropriate test fixture does not already exist that fits your test scenario then follow these steps to create
+one!
+
+#. Create a fixture that inherits from ``_zigzag_runner_factory`` and is scoped to live for the whole test session::
+
+    @pytest.fixture(scope='session')
+    def a_bunch_of_failures(_zigzag_runner_factory):
+        # code
+
+#. Invoke a ``ZigZagRunner`` from the factory with a **unique** file name along with the desired CI environment to emulate::
+
+    zz_runner = _zigzag_runner_factory('a_bunch_of_failures.xml', 'asc')
+
+#. Add test cases with desired attributes. (See ``TestCaseInfo`` for more details) ::
+
+    zz_runner.add_test_case('failed')
+    zz_runner.add_test_case('failed')
+    zz_runner.add_test_case('failed')
+
+#. Make sure to return the runner to the caller::
+
+    return zz_runner
+
+Building a Test Case with a Fixture
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Once a fixture has been selected/created then you need to inherit the fixture for you::
+
+    def test_a_bunch_of_failures(a_bunch_of_failures):
+        # code
+
+Invoking ZigZag
+^^^^^^^^^^^^^^^
+
+To invoke ``ZigZag`` use::
+
+    a_bunch_of_failures.assert_invoke_zigzag()
+
+Or use::
+
+    a_bunch_of_failures.invoke_zigzag()
+
+The suggested method to use is ``assert_invoke_zigzag`` because it will do automatic basic validation that the test
+case exists after invoking ``ZigZag``. The ``invoke_zigzag`` method should only be used when attempting to validate
+a negative scenario or more fine grained control is needed when validating state in qTest.
+
+Clean-up
+^^^^^^^^
+
+The ``_configure_test_environment`` and ``_zigzag_runner_factory`` private fixtures already perform clean-up duties
+during test run session teardown. However, sometimes it is necessary to clean-up artifacts during test case execution.
+The ``TestCaseInfo``, ``TestSuiteInfo`` and ``ZigZagRunner`` classes all provide a ``clean_up`` method which will
+remove artifacts. Use the ``clean_up`` with caution because misuse *could* cause teardown in upstream fixtures to fail
+in bizarre ways. Basically, these knives are sharp and they will cut you.
+
+Validation
+^^^^^^^^^^
+
+The ``TestCaseInfo``, ``TestSuiteInfo`` and ``ZigZagRunner`` classes all provide some basic ``assert`` methods for
+state validation. For complex test scenarios the state of artifacts in the qTest project under test will need to be
+done using the `qTest swagger client`_ by the develop. FYI the ``ZigZagRunner`` class provides properties that can
```

Comment:
I don't understand the intent of the "by the develop" clause. "by the developer"?? If so, can you provide a link or context for using the `qTest_swagger_client`?

nitpick: s/FYI/FYI,/

Reply:
Yeah, if you render the RST it will be a clickable link. The reference is at the bottom of the document.